### DISCUSSION TITLE
Limit the snapshot number when "pushing" to avoid creating dozens of thousands of snapshots

### DISCUSSION THREAD

**Discussion by wxiaoguang** [zrepl/zrepl#809] [Q&A] [answered]
My zrepl fails when pushing, but it still keeps creating new snapshots.

Now there are more than 98,000 snapshots on the machine and I have to manually destroy them one by one ...

(and I have no idea whether zrepl would be able to recover and continue pushing)

**Comment by problame** [comment, marked as answer]
Please post your zrepl config, esp the pruning rules.

I suspect you are using the `not_replicated` rule somewhere. Removing it should make zrepl clean them up for you (unless of course other pruning rules are holding on to those snapshots)

https://zrepl.github.io/configuration/prune.html#policy-not-replicated

**Comment by wxiaoguang** [comment]
Oh yes, thank you for pointing it out.

I guess it is safe to remove `- type: not_replicated` , will try to figure out.
